pydtk/io/writer.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright Toolkit Authors
+
+"""pydtk modules."""
+
+from abc import ABCMeta
+
+from pydtk.models import MODELS_BY_PRIORITY, MetaDataModel
+from pydtk.preprocesses import PassThrough
+
+
+class NoModelMatchedError(BaseException):
+    """Error of module matching."""
+
+    pass
+
+
+class BaseFileWriter(metaclass=ABCMeta):
+    """Base file writer."""
+
+    _model = None   # model used for loading a file
+
+    def __init__(self, **kwargs):
+        self.preprocesses = [PassThrough()]
+
+    @classmethod
+    def _select_model(cls, file_metadata):
+        """Select a proper model based on the given file-metadata.
+
+        Args:
+            file_metadata (object): an MetaDataModel object
+
+        """
+        priorities = MODELS_BY_PRIORITY.keys()
+        for priority in sorted(priorities, reverse=True):
+            for model in MODELS_BY_PRIORITY[priority]:
+                if model.is_loadable(**file_metadata.data):
+                    return model
+        raise NoModelMatchedError('No suitable model found for loading data: {}'.
+                                  format(file_metadata))
+
+    @property
+    def model(self):
+        """Return model."""
+        return self._model
+
+    @model.setter
+    def model(self, model):
+        self._model = model
+
+    def add_preprocess(self, preprocess):
+        """Add preprocessing function."""
+        self.preprocesses += [preprocess]
+
+    def write(self,
+              metadata=None,
+              data=None,
+              model_kwargs=None,
+              **kwargs
+              ):
+        """Write a file which corresponds to the given metadata.
+
+        Args:
+            metadata (MetaDataModel or dict): metadata of the data to save
+            data (numpy array): data
+            model_kwargs (dict): kwargs to pass to the selected model
+
+        Returns:
+            void
+
+        """
+        if model_kwargs is None:
+            model_kwargs = {}
+
+        # Check metadata is valid
+        if metadata is None:
+            raise ValueError('Metadata must be specified')
+        elif type(metadata) is not dict:
```

Comment:
Please accept type `MetaDataModel` as metadata or fix the corresponding docstring (line 66).

Reply:
I think fixing docstring is better.
Because io.writer is used in a situation that neither data and metadata exist in temporary memory only. If both data exists as a file, db.handler sould be used. In addition, I think only one side of data rarely exist as file ( and loading the data or saving another data could be easy by using model.load/save ). 
How do you think about?
（英語力に危機感があって，かつ，現状余裕があるので，練習がてらに英語で書いてみました．今後もこんな感じで行くつもりですが，業務上良くなかったり，英語が下手だったら教えてくださると幸いです）